packages/suite-native-atoms/src/Select/Select.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ReactNode, useState } from 'react';
+
+import { CryptoIcon, CryptoIconName, FlagIcon, FlagIconName } from '@trezor/icons';
+
+import { isCryptoIconType, isFlagIconType } from './selectTypes';
+
+import { BottomModal } from '../Modal/BottomModal';
+import { SelectTrigger } from './SelectTrigger';
+import { SelectItem } from './SelectItem';
+
+export type SelectItem = {
+    label: string;
+    value: string;
+    iconName?: FlagIconName | CryptoIconName;
+};
+
+type SelectProps = {
+    items: SelectItem[];
+    label: string;
+    selectedItem: SelectItem | null;
+    onSelectItem: (item: SelectItem | null) => void;
+};
+
+export const Select = ({ items, label, selectedItem, onSelectItem }: SelectProps) => {
+    const [isOpen, setIsOpen] = useState<boolean>(false);
+
+    const handleSelectItem = (value: string) => {
+        onSelectItem(items.find(item => item.value === value) ?? null);
+        setIsOpen(false);
+    };
+
+    const getIcon = (iconName?: CryptoIconName | FlagIconName, isSelectItem = false): ReactNode => {
+        if (!iconName) return null;
+        if (isCryptoIconType(iconName)) {
+            return <CryptoIcon size={isSelectItem ? 'small' : 'extraSmall'} name={iconName} />;
+        }
+        if (isFlagIconType(iconName)) {
+            return <FlagIcon size={isSelectItem ? 'small' : 'extraSmall'} name={iconName} />;
+        }
+    };
+
+    return (
+        <>
+            {isOpen && (
+                <BottomModal
+                    isVisible={isOpen}
+                    onVisibilityChange={setIsOpen}
+                    title="Typography Demo"
+                    hasBackArrow
+                    onBackArrowClick={() => setIsOpen(false)}
+                >
+                    {items.map(({ value, label, iconName }, index) => (
+                        <SelectItem
+                            key={value}
+                            label={label}
+                            value={value}
+                            icon={getIcon(iconName, true)}
+                            isSelected={value === selectedItem?.value}
+                            isLastChild={index === items.length - 1}
+                            onSelect={() => handleSelectItem(value)}
+                        />
+                    ))}
+                </BottomModal>
+            )}
+            <SelectTrigger
```

Comment:
why do you need pass `value` here? I think label should be enough

Reply:
Because if value is selected, it is shown in the select as "label"

Reply:
[b54e127](https://github.com/trezor/trezor-suite/pull/5506/commits/b54e12792e1a024ee488adf91ca238da4f65e391)